### DISCUSSION TITLE
Why cmake OPUS_FOUND needs OPUS_LIBRARY?

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2360] [Q&A]
I'm building static `libbaresip.a` lib and build does not find OPUS unless I include this cmake argument:
```
-DOPUS_LIBRARY=path_to_libopus.a
```
Why is `-DOPUS_INCLUDE_DIR` not enough and also `-DOPUS_LIBRARY` is required?  Isn't is enough that `libopus.a` is included when the app using `libbaresip.a` is built?

**Comment by juha-h** [comment]
I made an experiment:
```
$ git diff
diff --git a/cmake/FindOPUS.cmake b/cmake/FindOPUS.cmake
index fce4308d..175082e3 100644
--- a/cmake/FindOPUS.cmake
+++ b/cmake/FindOPUS.cmake
@@ -18,7 +18,7 @@ find_path(OPUS_INCLUDE_DIR
 )
 
 find_library(OPUS_LIBRARY
-  NAMES opus
+  NAMES ogg
   HINTS
     "${OPUS_LIBRARY_DIRS}"
     "${OPUS_HINTS}/lib"
```
and then built libbaresip.  After that I built my app than has in its CMakeLists.txt:
```
add_library(lib_opus STATIC IMPORTED)
set_target_properties(lib_opus PROPERTIES IMPORTED_LOCATION
        ${distribution_DIR}/opus/lib/${ANDROID_ABI}/libopus.a)
...
add_library(lib_baresip STATIC IMPORTED)
set_target_properties(lib_baresip PROPERTIES IMPORTED_LOCATION
        ${distribution_DIR}/baresip/lib/${ANDROID_ABI}/libbaresip.a)
```
And the resulting app worked fine.

Opus is just an example.  There are many modules and even libre.a and librem.a with the same issue:  what is really needed to build libbaresip.a?

**Comment by juha-h** [comment]
If opus functions used by `libbaresip.a` would be included in `libbaresip.a` then it would be OK to make build of `libbaresip.a` dependent on `libopus.a`.  But my understanding is that they are not included and that any app that is using `libbaresip.a` with opus, needs to have both `libbaresip.a` and `libopus.a` included in its build.  If I'm correct, this starts to look like a bug in baresip cmake build.  Should I convert this discussion into issue?

**Comment by alfredh** [comment]
libbaresip comes in two variants, regular or static

In regular build all modules are linked dynamically and libbaresip has no compile-time knowledge of them.

In static build all modules are linked together with libbaresip. This is needed for platforms that
do not provide loadable modules, such as Windows and iOS.

for cross compiling there is a special flag in the toolchain.cmake file that hides the path for FindXXX,
please have a look here:

https://github.com/baresip/baresip-win32/blob/main/cmake/mingw-w64-x86_64.cmake#L19

**Reply by juha-h** [reply]
My libbaresip-android build https://github.com/juha-h/libbaresip-android/blob/master/Makefile builds static libbaresip.a, but based on my experience, libopus.a (or other libs libbaresip.a depends on) are _not_  linked together with libbaresip.a.  Instead, I must add them in CMakeList.txt  https://github.com/juha-h/baresip-studio/blob/master/app/src/main/cpp/CMakeLists.txt that builds the baesip app.
